src/liger_kernel/transformers/monkey_patch.py

```diff
@@ -212,3 +216,54 @@ def apply_liger_kernel_to_phi3(
         modeling_phi3.Phi3MLP = LigerPhi3SwiGLUMLP
     if cross_entropy:
         modeling_phi3.CrossEntropyLoss = LigerCrossEntropyLoss
+
+
+# Model type corresponds to the keys defined in transformers/models/auto/modeling_auto.py
+MODEL_TYPE_TO_APPLY_LIGER_FN = {
+    "gemma": apply_liger_kernel_to_gemma,
+    "llama": apply_liger_kernel_to_llama,
+    "mistral": apply_liger_kernel_to_mistral,
+    "mixtral": apply_liger_kernel_to_mixtral,
+    "qwen2": apply_liger_kernel_to_qwen2,
+    "phi3": apply_liger_kernel_to_phi3,
+}
+
+
+def _apply_liger_kernel(model_type: str = "", **kwargs) -> None:
+    """
+    Applies Liger kernels based on the specified model type. The custom
+    kernels for the specified model type will be applied with the provided
+    keyword arguments, otherwise the default configuration will be used.
+
+    Args:
+        - model_type: the model types as defined in transformers/models/auto/modeling_auto.py
+          and specified in the model's config.json
+        - kwargs: keyword arguments that are passed to the corresponding apply_liger_kernel_to_* function.
+    """
+
+    if not model_type:
+        logger.info("Model type was not provided. No Liger kernels will be applied.")
+        return
+
+    if model_type not in MODEL_TYPE_TO_APPLY_LIGER_FN.keys():
```

Comment:
Correct me if I am wrong. This still doesn't fully resolve the original request in the issue "Liger is missing support for a number of language model architectures. Additionally, I'd like to be able to patch any model without consideration for the model architecture using a simple apply_liger_kernel_to_all_modules(pretrained_model)" where users want to automatically apply this to appropriate layers / modules if that module is identical to one of supported models.

Reply:
That's correct, discussing with @ByronHsu, comparing module code diff to determine if we can patch is a bit hacky, so want to hold off on that and just maintain the supported model list in the library. In any case, this will be a wrapper on top of any dynamic patching of models if we decide to implement in the future.

Also, this change doesn't provide a function to apply any and all liger kernels globally using default settings. I think that could be valuable so users could just globally set this at the beginning of their training script and not worry if they are using different AutoModels*/different model types. If we do, we can have it in a separate PR.

I don't believe there would be conflicts? For example, even though we throw an error if `fused_linear_cross_entropy` and `cross_entropy` are both set to True, I don't think it would actually cause a conflict to patch both of them. Thoughts @ByronHsu ?

Reply:
After discussion w steven. We want to have two high-level APIs now

1. existing per model patching
2. autoligerclass

which emphasizes on ease of use and convenience since this is the reason people like liger kernel.

for others, we can revisit based on community's demand. we don't want to push everything out at time or users will get confused.